test/components/preprocessors/test_csv_document_splitter.py

```diff
@@ -301,3 +317,20 @@ def test_from_dict_non_defaults(self) -> None:
         assert splitter.row_split_threshold == 1
         assert splitter.column_split_threshold is None
         assert splitter.read_csv_kwargs == {"sep": ";"}
+
+    def test_split_by_row(self, csv_with_four_rows: str) -> None:
+        splitter = CSVDocumentSplitter(split_by_row=True)
+        doc = Document(content=csv_with_four_rows)
+        result = splitter.run([doc])["documents"]
+        assert len(result) == 4
+        assert result[0].content == "A,B,C\n"
+        assert result[1].content == "1,2,3\n"
+        assert result[2].content == "X,Y,Z\n"
+
+    def test_split_by_row_with_empty_rows(self, caplog) -> None:
```

Comment:
Thanks for thinking of this scenario. I think instead of the warning be detected in `_split_by_row_mode` I think it would be good to move the check for an empty table in the `run` method right after reading the dataframe. 

Then we could also explicitly say in the warning message that the document is being skipped, similar to what we do in our PDF converters (e.g. PyPDF). Let's make sure to also add info about which Document was skipped (e.g. Doc ID) so users could identify the problematic document from the logs.